src/mpi/coll/transports/gentran/tsp_gentran.c

```diff
@@ -557,6 +562,7 @@ int MPIR_TSP_sched_sub_sched(MPIR_TSP_sched_t s, MPIR_TSP_sched_t subs,
     MPII_Genutil_vtx_add_dependencies(sched, *vtx_id, n_in_vtcs, in_vtcs);
     vtxp->u.sched.sched = subsched;
     vtxp->u.sched.req = NULL;
+    vtxp->u.sched.is_persistent = subsched->is_persistent;
```

Comment:
I think this and the previous commit make more sense squashed into the commit that adds the sub-scheduler vertex type.

Reply:
The previous change adding "req" to scheduler is part of that commit because the new vertex type added (for error checking) uses the "req" in the same commit. Moving req to later commit breaks it. I think even the commit to add schede_irecv_status technically speaking is also part of the error checking commit group.

And this change - propagating the is_persistent from a scheduler to the sub-scheduler vertex in the parent tree is also part of this commit, because otherwise, freeing the parent scheduler can accidentally freeing the subscheduler which is a persistent scheduler. This is exactly what this commit was trying to fix.

Reply:
There are 3 commits
```
coll: add a new Gentran vertex type to encapsulate a sub scheduler
coll/tsp: reset schedule function needs to reset sub-schedule
coll/tsp: free sub scheduler when the main scheduler is freed
```
My understanding is that the first adds a broken sub-schedule vertex that is subsequently fixed up by the next 2 commits. My suggestion is to squash them into a single, non-broken commit. Or is my understanding wrong?

Reply:
Got you. Yes, that makes sense.

Reply:
done